go/src/client_test.go

```diff
@@ -0,0 +1,274 @@
+/* Copyright (c) 2020 vesoft inc. All rights reserved.
+ *
+ * This source code is licensed under Apache 2.0 License,
+ * attached with Common Clause Condition 1.0, found in the LICENSES directory.
+ */
+
+package ngdb
+
+import (
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/vesoft-inc/nebula-clients/go/nebula/graph"
+	conf "github.com/vesoft-inc/nebula-clients/go/src/conf"
+	data "github.com/vesoft-inc/nebula-clients/go/src/data"
+	nebulaNet "github.com/vesoft-inc/nebula-clients/go/src/net"
+)
+
+const (
+	address  = "127.0.0.1"
+	port     = 34187
+	username = "user"
+	password = "password"
+)
+
+var poolAddress = []*data.HostAddress{
+	&data.HostAddress{
+		Host: "127.0.0.1",
+		Port: 3699,
+	},
+	&data.HostAddress{
+		Host: "127.0.0.1",
+		Port: 3701,
+	},
+	&data.HostAddress{
+		Host: "127.0.0.1",
+		Port: 3710,
+	},
+}
+
+// Create default configs
+var testPoolConfig = conf.NewPoolConf(0, 0, 0, 0, 0)
+
+// Before run `go test -v`, you should start a nebula server listening on 3699 port.
+// Using docker-compose is the easiest way and you can reference this file:
+//   https://github.com/vesoft-inc/nebula/blob/master/docker/docker-compose.yaml
+//
+// TODO(yee): introduce mock server
+
+func logoutAndClose(conn *nebulaNet.Connection, sessionID int64) {
+	conn.SignOut(sessionID)
+	conn.Close()
+}
+
+func TestConnection(t *testing.T) {
+	if testing.Short() {
+		t.Skip("Skipping client test in short mode")
+	}
+
+	hostAdress := data.NewHostAddress(address, port)
+
+	conn := nebulaNet.NewConnection(hostAdress)
+	err := conn.Open(hostAdress, testPoolConfig)
+	if err != nil {
+		t.Fatalf("Fail to open connection, address: %s, port: %d, %s", address, port, err.Error())
+	}
+
+	authresp, authErr := conn.Authenticate(username, password)
+	if authErr != nil {
+		t.Fatalf("Fail to authenticate, username: %s, password: %s, %s", username, password, authErr.Error())
+	}
+
+	sessionID := authresp.GetSessionID()
+
+	defer logoutAndClose(conn, sessionID)
+
+	checkResp := func(prefix string, authresp *graph.ExecutionResponse) {
```

Comment:
execResp

Reply:
checkResp seems to be a more appropriate name for this function